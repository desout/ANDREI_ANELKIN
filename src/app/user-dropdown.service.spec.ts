import {TestBed} from '@angular/core/testing';
import { UserDropDownService } from './user-dropdown.service';
import {TestModule} from './Test.module';

describe('UserDropDownService', () => {
  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [TestModule]
    });
  });
  it('getDropDownUser: Should return User with correct age', () => {
    const mockUser = {
      id: 0,
      age: 21,
      name: 'Andrei',
      password: 'drogba',
      dateOfBirth: '1997/10/23',
      dateOfFirstLogin: '2017/10/23',
      dateNextNotification: '2018/10/23',
      information: 'Best of the Best',
      role: 'ADMIN'
    };
    const service: UserDropDownService = TestBed.get(UserDropDownService);

    const user = service.getDropDownUser({
      id: 0,
      name: 'Andrei',
      password: 'drogba',
      dateOfBirth: '1997/10/23',
      dateOfFirstLogin: '2017/10/23',
      dateNextNotification: '2018/10/23',
      information: 'Best of the Best',
      role: 'ADMIN'});

    expect(user).toEqual(mockUser);
    expect(user).toMatchSnapshot();
  });
  it('getAge: should return correct age', () => {
    const age = 19;
    const service: UserDropDownService = TestBed.get(UserDropDownService);

    const letAge = service.getAge('1999/10/23');

    expect(letAge).toEqual(age);
    expect(letAge).toMatchSnapshot();
  });
});
