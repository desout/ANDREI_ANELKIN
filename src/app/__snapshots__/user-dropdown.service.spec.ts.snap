// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDropDownService getAge: should return correct age 1`] = `19`;

exports[`UserDropDownService getDropDownUser: Should return User with correct age 1`] = `
Object {
  "age": 21,
  "dateNextNotification": "2018/10/23",
  "dateOfBirth": "1997/10/23",
  "dateOfFirstLogin": "2017/10/23",
  "id": 0,
  "information": "Best of the Best",
  "name": "Andrei",
  "password": "drogba",
  "role": "ADMIN",
}
`;
